### DISCUSSION TITLE
aws secret manager

### DISCUSSION THREAD

**Discussion by houshmandX** [nextgenhealthcare/connect#4968] [Q&A]
Hi,

I'm trying to use secret manager for mirthconnect for my DB connections. AWS use "var AWS = require('aws-sdk')" in order to access aws SDK, but we can not use "require()" in mirthconnect. 

One way I did is to create a MAVEN project and use aws dependencies, and use java file for the AWS secret manager and create a .jar file and use that .jar file in mirth. But the file I have (java file) compiles and run on my local, but the issue is when I try to run .jar file and call the package and class, in my local, and in mirth, I get error regarding the libraries are unknown!

  <modelVersion>4.0.0</modelVersion>
  <groupId>aws.getsecretmanager</groupId>
  <artifactId>aws-get-secret</artifactId>
  <version>1.0-SNAPSHOT</version>
  <packaging>jar</packaging>
<dependency>
      <groupId>com.amazonaws</groupId>
      <artifactId>aws-java-sdk-core</artifactId>
      <version>1.11.325</version>
    </dependency>
    <dependency>
      <groupId>com.amazonaws</groupId>
      <artifactId>aws-java-sdk-secretsmanager</artifactId>
      <version>1.11.339</version>
    </dependency>
.
.
.
**********

package aws.getsecretmanager;

import java.io.IOException;
import java.nio.ByteBuffer;
//import java.nio.charset.Charset;

import com.amazonaws.AmazonClientException;
import com.amazonaws.AmazonServiceException;
import com.amazonaws.client.builder.AwsClientBuilder;
//import com.amazonaws.regions.Regions;
import com.amazonaws.services.secretsmanager.AWSSecretsManager;
import com.amazonaws.services.secretsmanager.AWSSecretsManagerClientBuilder;
import com.amazonaws.services.secretsmanager.model.GetSecretValueRequest;
import com.amazonaws.services.secretsmanager.model.GetSecretValueResult;
import com.amazonaws.services.secretsmanager.model.ResourceNotFoundException;
import com.amazonaws.services.secretsmanager.model.InvalidRequestException;
import com.amazonaws.services.secretsmanager.model.InvalidParameterException;

public class SecretManagerSample {
    public static void getSecret(String secretName, String region) {
        String endpoint =("secretsmanager." + region + ".amazonaws.com");
        AwsClientBuilder.EndpointConfiguration config = new AwsClientBuilder.EndpointConfiguration(endpoint, region);
        AWSSecretsManagerClientBuilder clientBuilder = AWSSecretsManagerClientBuilder.standard();
        clientBuilder.setEndpointConfiguration(config);
        AWSSecretsManager client = clientBuilder.build();
        String secret = null;
        ByteBuffer binarySecretData;
        GetSecretValueRequest getSecretValueRequest = new GetSecretValueRequest().withSecretId(secretName);
        GetSecretValueResult getSecretValueResponse = null;
        try {
            getSecretValueResponse = client.getSecretValue(getSecretValueRequest);

        } catch(ResourceNotFoundException e) {
            System.out.println("The requested secret " + secretName + " was not found");
        } catch (InvalidRequestException e) {
            System.out.println("The request was invalid due to: " + e.getMessage());
        } catch (InvalidParameterException e) {
            System.out.println("The request had invalid params: " + e.getMessage());
        }

        if(getSecretValueResponse == null) {
            return;
        }

        // Decrypted secret using the associated KMS CMK
        // Depending on whether the secret was a string or binary, one of these fields will be populated
        if(getSecretValueResponse.getSecretString() != null) {
            secret = getSecretValueResponse.getSecretString();
        }
        else {
            binarySecretData = getSecretValueResponse.getSecretBinary();
        }
    }
    public static void main(String[] args) throws IOException {
       .......
    }
}

*******************
$ mvn package

-this creates a .jar file 
$java -cp target/aws-get-secret-1.0-SNAPSHOT.jar aws.getsecretmanager.SecretManagerSample
getting error:
Error: Unable to initialize main class aws.getsecretmanager.SecretManagerSample
Caused by: java.lang.NoClassDefFoundError: com/amazonaws/AmazonServiceException

******************

Can someone help me with this issue?

Thank you!

**Comment by ntjoumas** [comment]
Did you ever get this resolved?

**Reply by houshmandX** [reply]
yes, I have resolved it.

**Reply by pacmano1** [reply]
LOL, I'd imagine @ntjoumas is probably also asking "how did you resolve it"?

**Reply by ntjoumas** [reply]
That would be a correct statement @pacmano1 :)

**Reply by houshmandX** [reply]
lol, magic. Followed this amazon link, and I used pretty much the same dependencies and java, and it did work for me. 

https://docs.aws.amazon.com/code-samples/latest/catalog/java-secretsmanager-pom.xml.html
